activation/handler_v2.go

```diff
@@ -0,0 +1,177 @@
+package activation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/spacemeshos/post/shared"
+	"go.uber.org/zap"
+
+	"github.com/spacemeshos/go-spacemesh/activation/wire"
+	"github.com/spacemeshos/go-spacemesh/atxsdata"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/datastore"
+	"github.com/spacemeshos/go-spacemesh/log"
+	mwire "github.com/spacemeshos/go-spacemesh/malfeasance/wire"
+	"github.com/spacemeshos/go-spacemesh/p2p"
+	"github.com/spacemeshos/go-spacemesh/signing"
+	"github.com/spacemeshos/go-spacemesh/sql/atxs"
+	"github.com/spacemeshos/go-spacemesh/system"
+)
+
+//go:generate mockgen -typed -source=./handler_v2.go -destination=mocks_handler_v2.go -package=activation
+
+type nipostValidatorV2 interface {
+	IsVerifyingFullPost() bool
+	VRFNonceV2(smesherID types.NodeID, commitment types.ATXID, vrfNonce types.VRFPostIndex, numUnits uint32) error
+	PostV2(
+		ctx context.Context,
+		smesherID types.NodeID,
+		commitment types.ATXID,
+		post *wire.PostV1,
+		challenge []byte,
+		numUnits uint32,
+		opts ...validatorOption,
+	) error
+}
```

Comment:
Shall we put this into `interface.go`?

Reply:
I think not. I'm leaning towards defining the interface next to the thing that uses it and that's needed for the ATX Handler V2 only. I want to create a similar interface for the Handler V1 and maybe have something like this in the `handler.go` (and remove `nipostValidator` interface from interface.go entirely).:

```go
type nipostValidator interface {
	nipostValidatorV1
	nipostValidatorV2
}
```

The
```go
VerifyChain(ctx context.Context, id, goldenATXID types.ATXID, opts ...VerifyChainOption) error
```

shouldn't be in the nipostValidator in the first place most likely.

Wdyt?

Reply:
Smaller interfaces are generally better yes and I'm OK with your suggestion.